### PR TITLE
[specfile] new fedora guidelines

### PR DESCRIPTION
**Pull request by sandrobonazzola**
Adapt the specfile to new fedora packaging guidelines.

- use pyproject macros (see also https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_example_spec_file )
- use bin instead of sbin (see also [rhbz#2210708](https://bugzilla.redhat.com/show_bug.cgi?id=2210708)

Based on top of PR #3259

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by TurboTurtle** [comment]
I'm assuming we want to use this one over #3159?

Beyond that - I'm taking this to mean we need to install `python3-wheel` on the builder VMs?
```
error: Failed build dependencies:
	python3dist(wheel) is needed by sos-4.5.4-1.fc37.noarch
```

If so, that's fine - it'd just need to be added to the setup script for cirrus - here: https://github.com/sosreport/sos/blob/main/.cirrus.yml#L103

**Comment by sandrobonazzola** [comment]
This one is rebased on top of #3259 ; the reason I pushed it as separate PR is because it introduces a breaking change moving the scripts from `sbin` to `bin` (on Fedora) and I wanted to be sure you're ok with the change.
I'll add the missing package into `cirrus.yml`

**Review by pmoravec — APPROVED** [review]

**Comment by pmoravec** [comment]
Sigh, I should review failed tests prior ACKing a PR :)

CentOS Stream 9 does not contain `python3-wheel` and Fedora build fails on:

    error: Failed build dependencies:
	    python3dist(pexpect) is needed by sos-4.5.4-1.fc38.noarch
    	    python3dist(pyyaml) is needed by sos-4.5.4-1.fc38.noarch

that I dont understand why it can happen..?

**Comment by TurboTurtle** [comment]
> This one is rebased on top of #3259 ; the reason I pushed it as separate PR is because it introduces a breaking change moving the scripts from `sbin` to `bin` (on Fedora) and I wanted to be sure you're ok with the change. I'll add the missing package into `cirrus.yml`


I suspect most/all downstreams roll their own spec files, and in that regard Fedora is our "closest" downstream, so I'm fine with the stock spec file to be mainly geared towards Fedora guidelines, as long as it can still be used for ad-hoc builds on non-Fedora rpm boxes.

In other words: totally cool to change to `bin`.

**Comment by hroncok** [comment]
Is it intentional that RHEL 10 will not use this?

**Comment by sandrobonazzola** [comment]
> Is it intentional that RHEL 10 will not use this?

It wasn't my intention, but I'm not the RHEL package maintainer.

**Comment by hroncok** [comment]
I've noticed the package is built the new way in Fedora 39+ and the old way in Fedora ELN. I've tracked the origin of this conditional here, so I thought I'd ask. (I don't really care that much, IMHO using the new Python guidelines conditionally kinda eliminates their benefit anyway.)

**Comment by sandrobonazzola** [comment]
> I've noticed the package is built the new way in Fedora 39+ and the old way in Fedora ELN. I've tracked the origin of this conditional here, so I thought I'd ask. (I don't really care that much, IMHO using the new Python guidelines conditionally kinda eliminates their benefit anyway.)

The condition on fedora >= 39 is because it failed to build on fedora 38 with the new macros. And as RHEL 9 and 8 are using completely different spec file I didn't care checking eln/el9 for this. I'll take a look at next  rebase

**Pull request merged**
